packages/watch-eth/test/ethereum-block-processor.spec.ts

```diff
@@ -0,0 +1,101 @@
+import './setup'
+
+/* External Imports */
+import { Block } from 'ethers/providers'
+import { DB, getLogger, newInMemoryDB, sleep } from '@pigi/core'
+import { createMockProvider, deployContract, getWallets } from 'ethereum-waffle'
+
+/* Internal Imports */
+import { deployTokenContract, TestListener } from './utils'
+import { EthereumBlockProcessor } from '../src'
+
+const log = getLogger('ethereum-block-processor-test', true)
+
+const timeout = 25_000
+describe('Block Subscription', () => {
+  let provider
+  let wallets
+  let ownerWallet
+  let recipientWallet
+  let db: DB
+
+  const sendAmount = 100
+  const initialSupply = 100_000
+
+  let blockProcessor: EthereumBlockProcessor
+  let blockListener: TestListener<Block>
+  let tokenContract
+
+  beforeEach(async () => {
+    provider = createMockProvider()
+    wallets = getWallets(provider)
+    ownerWallet = wallets[0]
+    recipientWallet = wallets[1]
+
+    log.debug(`Connection info: ${JSON.stringify(provider.connection)}`)
+
+    tokenContract = await deployTokenContract(ownerWallet, initialSupply)
+
+    db = newInMemoryDB()
+    blockProcessor = new EthereumBlockProcessor(db)
+    blockListener = new TestListener<Block>()
+  })
+
+  it('processes new blocks', async () => {
+    await blockProcessor.subscribe(provider, blockListener, false)
+
+    await tokenContract.transfer(
+      ownerWallet.address,
+      recipientWallet.address,
+      sendAmount
+    )
+
+    const blocks: Block[] = await blockListener.waitForReceive(1)
+
+    blocks.length.should.equal(1)
```

Comment:
This can never be less than one, it can only be 1 or more (based on the `waitForReceive()` function) right?

Reply:
Correct!